src/coreclr/jit/emit.cpp

```diff
@@ -5078,6 +5079,52 @@ bool emitter::emitEndsWithAlignInstr()
     return emitCurIG->endsWithAlignInstr();
 }
 
+//-----------------------------------------------------------------------------
+//  emitRemoveIGAlignFlag: Remove the align flag of `alignInstr`.
+//
+void emitter::emitRemoveIGAlignFlag(instrDescAlign* alignInstr)
+{
+    emitLoopAlignRemoved = true;
+    alignInstr->removeAlignFlags();
+}
+
+//-----------------------------------------------------------------------------
+//  emitIGHadAlignFlag: Checks if 'igToCheck' ever was marked as aligned and later
+//                      decided to not align. Sometimes, a loop is marked as not
+//                      needing alignment, but the igSize was not adjusted immediately.
+//                      This method would helpful to know that specially during loopSize
+//                      calculation where we would adjust the loopsize by removed alignment
+//                      bytes.
+//
+//  Returns:  true iff igToCheck had alignment but was later removed. For any other IG
+//            whether was marked aligned or not, the method will return false.
+//
```

Comment:
Couldn't you just have a persistent flag on an IG that marks it as having ever been marked as needing alignment, so you wouldn't need to search the list?

And if you remove alignment from a block, shouldn't you remove it from the list?

Reply:
> And if you remove alignment from a block, shouldn't you remove it from the list?

We will set `IGF_REMOVED_ALIGN` only during `emitSetLoopBackEdge()`. At that time, we don't remove the `alignInstr` from the list nor do we adjust the `igSize` and `igOffs`. We do it later during `emitLoopAlignAdjustments()` phase. If we remove the `alignInstr` from the list, we will not process it and won't know that we need to readjust the offset and size of IGs that follows this one.